File: app/schemas/hypothesis.py
"""Pydantic response schemas for hypotheses."""

import uuid
from typing import List, Optional
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class HypothesisResponse(BaseModel):
    id: uuid.UUID
    title: str
    motivation: str
    core_claim: str
    method_sketch: str
    expected_outcome: str
    risk_factors: List[str]
    novelty_score: float
    feasibility_score: float
    hardware_requirement: str
    source_paper_ids: List[str]
    gap_description: str
    status: str
    iteration_count: int
    # Phase 3 Fields
    debate_rounds: Optional[int] = None
    arbiter_notes: Optional[str] = None
    approved_at: Optional[datetime] = None
    rejection_reason: Optional[str] = None
    parent_hypothesis_id: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class HypothesisListResponse(BaseModel):
    items: List[HypothesisResponse]
    total: int

File: scientific-copilot-ui/lib/types/hypothesis.ts
export type HypothesisStatus = 'pending' | 'approved' | 'rejected' | 'running' | 'done'

export interface Hypothesis {
  id: string
  title: string
  motivation: string
  core_claim: string
  method_sketch: string
  expected_outcome: string
  risk_factors: string[]
  novelty_score: number
  feasibility_score: number
  hardware_requirement: string
  source_paper_ids: string[]
  gap_description: string
  status: HypothesisStatus
  iteration_count: number
  parent_hypothesis_id: string | null
  approved_at: string | null
  rejection_reason: string | null
  debate_rounds: number | null
  arbiter_notes: string | null
  created_at: string
  updated_at: string
}

export interface PaginatedHypotheses {
  items: Hypothesis[]
  total: number
}

export interface DebateResult {
  hypothesis_id: string
  verdict: string
  rounds: number
}

export interface DebateSummary {
  debated: number
  passed: number
  failed: number
}
